src/main/scala/com/virtuslab/shared_indexes/core/SharedIndexes.scala

```diff
@@ -2,70 +2,12 @@ package com.virtuslab.shared_indexes.core
 
 object SharedIndexes {
 
-  def dumpJdkSharedIndexes(
```

Comment:
I'd rather keep these methods here in one place, because they are essentially overloads of the same method.

Reply:
As we have a generator for each kind, I think these classes are the right palce to put the logic relevant to their kind. Maybe even the common method could go to the base class or it's companion.
IMO the project would be more modular if we organized classes more like this

core
jdk
jar
project
app

it would make classes in each package more coherent and allow to make most of them package private for example.